tests/phpunit/tests/date/wpDate.php

```diff
@@ -61,4 +61,121 @@ public function test_should_keep_localized_slashes() {
+		$utc = new DateTimeZone( 'UTC' );
+		$this->assertSame( (string) time(), wp_date( 'U', null, $utc ) );
+	}
+
+	/**
+	 * Tests that the date is formatted with no timezone provided.
+	 *
+	 * @ticket 53485
+	 */
+	public function test_should_format_date_with_no_timezone() {
+		$utc      = new DateTimeZone( 'UTC' );
+		$datetime = new DateTimeImmutable( '2019-10-17', $utc );
+		$this->assertSame( 'October', wp_date( 'F', $datetime->getTimestamp() ) );
+	}
+
+	/**
+	 * Tests that the format is set correctly.
+	 *
+	 * @dataProvider data_should_format_date
+	 *
+	 * @ticket 53485
+	 *
+	 * @param string $expected The expected result.
+	 * @param string $format   The date format.
+	 */
+	public function test_should_format_date( $expected, $format ) {
+		$utc      = new DateTimeZone( 'UTC' );
+		$datetime = new DateTimeImmutable( '2019-10-17', $utc );
+
+		$this->assertSame( $expected, wp_date( $format, $datetime->getTimestamp(), $utc ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_should_format_date() {
+		return array(
+			'Swatch Internet Time'                        => array(
+				'expected' => '041',
+				'format'   => 'B',
+			),
+			'Ante meridiem and Post meridiem (uppercase)' => array(
+				'expected' => 'AM',
+				'format'   => 'A',
+			),
+			'Ante meridiem and Post meridiem (uppercase) and escaped "A"' => array(
+				'expected' => 'A AM',
+				'format'   => '\\A A',
+			),
+			'Ante meridiem and Post meridiem (lowercase)' => array(
+				'expected' => 'am',
+				'format'   => 'a',
+			),
+			'Month'                                       => array(
+				'expected' => 'October',
+				'format'   => 'F',
+			),
+			'Month (abbreviated'                          => array(
+				'expected' => 'Oct',
+				'format'   => 'M',
+			),
+			'Weekday'                                     => array(
+				'expected' => 'Thursday',
+				'format'   => 'l',
+			),
+			'Weekday (abbreviated)'                       => array(
+				'expected' => 'Thu',
+				'format'   => 'D',
+			),
+		);
+	}
+
+	/**
+	 * Tests that the date is formatted when
+	 * `$wp_locale->month` and `$wp_locale->weekday` are empty.
+	 *
+	 * @ticket 53485
+	 */
+	public function test_should_format_date_with_empty_wp_locale_month_and_weekday() {
+		global $wp_locale;
+
+		$utc      = new DateTimeZone( 'UTC' );
+		$datetime = new DateTimeImmutable( '2019-10-17', $utc );
+
+		$original_wp_locale = $wp_locale;
+		$wp_locale->month   = array();
+		$wp_locale->weekday = array();
+		$actual             = wp_date( 'F', $datetime->getTimestamp(), $utc );
+		$wp_locale          = $original_wp_locale;
+
+		$this->assertSame( 'October', $actual );
+	}
+
+	/**
+	 * Tests the wp_date filter.
+	 *
+	 * @ticket 53485
+	 */
+	public function test_should_apply_filters_for_wp_date() {
+		add_filter(
+			'wp_date',
+			static function() {
+				return 'filtered';
+			}
+		);
+
+		$this->assertSame( 'filtered', wp_date( '' ) );
```

Comment:
This can use `MockAction` 

```suggestion
		$ma = new MockAction();
		add_filter( 'wp_date', array( &$ma, 'filter' ) );

		$this->assertSame( 1, $ma->get_call_count() );
```

Reply:
Nice. Added in [a6f80f9](https://github.com/WordPress/wordpress-develop/pull/2643/commits/a6f80f92f80e8b697a7630893a023ee49c228123).